### PR TITLE
[Bi-Directional, OPC UA, HTTP] backend implementation

### PR DESCRIPTION
**Pull request by Florian-Limpoeck**

### CODE REVIEW

File: hivemq-edge/src/main/java/com/hivemq/protocols/ProtocolAdapterWrapper.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
 import com.hivemq.extension.sdk.api.annotations.NotNull;
 import com.hivemq.extension.sdk.api.annotations.Nullable;
 
-public class ProtocolAdapterWrapper<T extends ProtocolAdapter> implements ProtocolAdapter {
```
Comment (DC2-DanielKrueger) [inline hivemq-edge/src/main/java/com/hivemq/protocols/ProtocolAdapterWrapper.java:32]: nice, was a big mistake to make it an ProtocolAdapter. Was easy at the time, but super error prone

**Review by codepitbull — CHANGES_REQUESTED** [review]
Tests broken

File: hivemq-edge/src/test/java/com/hivemq/protocols/ProtocolAdapterManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+        verify(remoteService).fireUsageEvent(any());
+        verify(protocolAdapterWritingService).stopWriting(adapterWrapper.getAdapter());
+    }
+
+    @Test
+    void test_startWriting_eventServiceFailedStart_resourcesCleanedUp() {
+
+        when(protocolAdapterWritingService.writingEnabled()).thenReturn(true);
+        when(protocolAdapterWritingService.startWriting(any(),
+                protocolAdapterWrapper.getProtocolAdapterMetricsService())).thenReturn(CompletableFuture.completedFuture(null));
```
[the remark this answers is not in the document]
Comment (Florian-Limpoeck) [reply hivemq-edge/src/test/java/com/hivemq/protocols/ProtocolAdapterManagerTest.java:181]: yeah it doesnt even compile, was a refactoring miss

**Review by codepitbull — APPROVED** [review]

**Review by DC2-DanielKrueger — APPROVED** [review]

File: modules/hivemq-edge-module-file/src/main/java/com/hivemq/edge/adapters/file/FileProtocolAdapterInformation.java [hunk 3]
```diff
@@ -46,7 +46,7 @@ protected FileProtocolAdapterInformation() {
 
     @Override
     public @NotNull String getProtocolId() {
```
Comment (DC2-DanielKrueger) [inline modules/hivemq-edge-module-file/src/main/java/com/hivemq/edge/adapters/file/FileProtocolAdapterInformation.java:48]: that breaks old configs as the protocol id is used in the config tag IIRC

**Pull request merged**